05_iigo/index.tex

```diff
@@ -528,11 +528,33 @@ \subsubsection{Server-side}
     \item The outcome of the action, or lack thereof, is passed to the members of the archipelago.
 \end{enumerate}
 
 \subsection{Legislative Branch}
+Similarly to the president the Speaker also has client-side functions which directly map to the Speaker powers described in \ref{sec:legislative}, as well as server-side functions which not only facilitate the implementation of these powers but also handle other functionality, such as the logging needed for monitoring and withdrawal of costs associated with the power.
+
+\subsubsection{Client-side}
+The legislative branch functions can be divided into two categories:
+\begin{enumerate}
+\item Speaker power functions:
+    \begin{itemize}
+        \item \texttt{DecideAgenda} does not directly map to a power of the speaker, but it does aid the agents with keeping track of the rule propagated through the legislative branch functions. By changing the rule given to the agent in \texttt{DecideAgenda}, which is the same rule returned by the President in \texttt{PickRuleToVote}, they can alter the rule passed to them in \texttt{DecideVote} and \texttt{DecideAnnouncement}.
+        \item \texttt{DecideVote} enables the institutional power of calling a vote and deciding the participating islands. The information passed to the Speaker is the current rule in the agenda (as returned by \texttt{DecideAgenda}) and a list of all the alive island IDs. The Speaker returns the same type of information along with a boolean indicating whether the agent has chosen to act on this power. The rule returned by the Speaker is the rule that is passed to agents returned by the Speaker if a vote is to be conducted as indicated by the boolean returned by the Speaker.
+        \item \texttt{DecideAnnouncement} enables the institutional power of declaring a result of a vote on rules. The information passed to the Speaker is the current rule in the agenda (as returned by \texttt{DecideAgenda}) and the result of the vote held. The Speaker returns the same type of information along with a boolean indicating whether the agent has chosen to act on this power. The rule returned by the Speaker along with the result returned by the Speaker is the information that is processed by the server and broadcasted to all the agents if the agent has chosen to make a declaration as indicated by the boolean returned by the Speaker.
+    \end{itemize}
+\item The accountability cycle and transfer-of-power functions:
+    \begin{itemize}
+        \item \texttt{PaySpeaker} implements the power of the Speaker to pay the Judge its salary. 
+        \item \texttt{CallJudgeElection} implements the power of the Speaker to call an election for the transfer-of-power for the Judge
+        \item \texttt{DecideNextJudge} allows the Speaker to modify the outcome of the election, if one is called, and choose not to follow its obligation (if Rule "TODO WHEN PR 4 IS MERGED" is in play) to announce the result of Judge election.
```

Comment:
can you change is TODO plz

Reply:
reference rule properly plz..
`Rule~\ref{rule:XX}` or smth like that should work